### PR TITLE
fix(permission): forward action approval exception for return expected response

### PR DESCRIPTION
**Pull request by nicolasalexandre9**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: app/services/forest_liana/ability/permission.rb [hunk 1]
```diff
@@ -35,13 +37,16 @@ def is_smart_action_authorized?(user, collection, parameters, endpoint, http_met
 
         user_data = get_user_data(user['id'])
         collections_data = get_collections_permissions_data
+        collection_name = ForestLiana.name_for(collection)
```
Comment (matthv) [inline app/services/forest_liana/ability/permission.rb:40]: 🥇

File: app/services/forest_liana/ability/permission.rb [hunk 2]
```diff
@@ -35,13 +37,16 @@ def is_smart_action_authorized?(user, collection, parameters, endpoint, http_met
...
+          action = find_action_from_endpoint(collection_name, endpoint, http_method).name
 
-          smart_action_approval = SmartActionChecker.new(parameters, collection, collections_data[ForestLiana.name_for(collection)][:actions][action], user_data)
+          smart_action_approval = SmartActionChecker.new(parameters, collection, collections_data[collection_name][:actions][action], user_data)
           smart_action_approval.can_execute?
+        rescue ForestLiana::Errors::ExpectedError => exception
+          raise exception
         rescue
-          raise ForestLiana::Errors::ExpectedError.new(409, :conflict, "The collection #{collection} doesn't exist", 'collection not found')
+          raise ForestLiana::Ability::Exceptions::UnknownCollection.new(collection_name)
```
Comment (matthv) [inline app/services/forest_liana/ability/permission.rb:49]: 👍

File: app/services/forest_liana/ability/permission.rb [hunk 3]
```diff
@@ -66,7 +71,7 @@ def is_chart_authorized?(user, parameters)
 
       private
 
-      def get_user_data(user_id)
+      def get_user_data(user_id, force_fetch = true)
```
Comment (matthv) [inline app/services/forest_liana/ability/permission.rb:74]: 👍

**Review by matthv — APPROVED** [review]

**Review by SteveBunlon — APPROVED** [review]
🏌️       ⛳

**Pull request merged**